### ISSUE TITLE
Wrong ordering on Index page

### ISSUE THREAD

**Issue by ikalnitsky** [ikalnitsky/holocron#169]
Steps to reproduce:

1. Create three posts, for instance:

   * **Feb 03, 2015** - Docker For Pythonistas
   * **Jun 14, 2015** - Mad Map: Python Road
   * **Oct 25, 2014** - C Legacy Is Evil

2. Build blog, and check out index page.

Expected result:

The **2015** archive contains two posts in the following order:

  * **Jun 14, 2015** - Mad Map: Python Road
  * **Feb 03, 2015** - Docker For Pythonistas

Actual result: 

The **2015** archive contains two posts in the following order:

  * **Feb 03, 2015** - Docker For Pythonistas
  * **Jun 14, 2015** - Mad Map: Python Road

So basically, we need to sort posts using `published` attribute in `desc` order.
